### ISSUE TITLE
API http post on balance status fails serialisation

### ISSUE THREAD

**Issue by phillxnet** [rockstor/rockstor-core#2396]
When executing a POST on balance status (empty data) we normally receive the last available PoolBalance record. However if no such balance record exists we return a 500 Exception.

## Reproducer:
via browser:
https://rleap15-3.lan/api/pools/1/balance/status (POST with default)
via Curl:
curl --insecure -u username:pass -X POST https://rleap15-3.lan/api/pools/1/balance/status

When the associated pool id has no balance records.

```
Exception: Got AttributeError when attempting to get a value for field `pool` on serializer `PoolBalanceSerializer`.
The serializer field might be named incorrectly and not match any attribute or key on the `Response` instance.
Original exception text was: 'Response' object has no attribute 'pool'.
Traceback (most recent call last):
  File "/opt/rockstor-dev/src/rockstor/rest_framework_custom/generic_view.py", line 41, in _handle_exception
    yield
  File "/opt/rockstor-dev/src/rockstor/storageadmin/views/pool_balance.py", line 225, in post
    return Response(PoolBalanceSerializer(ps).data)
  File "/opt/rockstor-dev/eggs/djangorestframework-3.9.3-py2.7.egg/rest_framework/serializers.py", line 563, in data
    ret = super(Serializer, self).data
  File "/opt/rockstor-dev/eggs/djangorestframework-3.9.3-py2.7.egg/rest_framework/serializers.py", line 262, in data
    self._data = self.to_representation(self.instance)
  File "/opt/rockstor-dev/eggs/djangorestframework-3.9.3-py2.7.egg/rest_framework/serializers.py", line 517, in to_representation
    attribute = field.get_attribute(instance)
  File "/opt/rockstor-dev/eggs/djangorestframework-3.9.3-py2.7.egg/rest_framework/relations.py", line 192, in get_attribute
    return super(RelatedField, self).get_attribute(instance)
  File "/opt/rockstor-dev/eggs/djangorestframework-3.9.3-py2.7.egg/rest_framework/fields.py", line 474, in get_attribute
    raise type(exc)(msg)
AttributeError: Got AttributeError when attempting to get a value for field `pool` on serializer `PoolBalanceSerializer`.
The serializer field might be named incorrectly and not match any attribute or key on the `Response` instance.
Original exception text was: 'Response' object has no attribute 'pool'.
```

This is because we are attempting to serialise a PoolBalance object that is not an instance of PoolBalance as there isn't a record of a pool balance (PoolBalance is our model for these). We already return Response() from:
storageadmin/views/pool_balance.py_balance_status(pool) but we then fail to recognise this as a non PoolBalance in our POST section that deals with balance status.

It is proposed that we instead return an empty Response() resulting in an http 200 when we POST to the status command.

Normally a post would return the last PoolBalance record that exists i.e. for a pool with 2 balance records with id=5 and id=6 we get:
```
curl --insecure -u admin:admin -X POST https://rleap15-3.lan/api/pools/3/balance/status
{"id":6,"status":"finished","tid":"d44eef1f-1c36-4515-8bc2-c1bbe966789a","message":null,"start_time":"2022-07-11T19:54:20.838200+01:00","end_time":"2022-07-11T19:54:25.954726+01:00","percent_done":100,"internal":false,"pool":3}
```
so for no records an empty response seems appropriate. I.e.
```
curl --insecure -u admin:admin -X POST https://rleap15-3.lan/api/pools/1/balance/status
```
Where as our current suspected inconsistent response for a pool with no balance records is:
```
curl --insecure -u admin:admin -X POST https://rleap15-3.lan/api/pools/1/balance/status
["Got AttributeError when attempting to get a value for field `pool` on serializer `PoolBalanceSerializer`.\nThe serializer field might be named incorrectly and not match any attribute or key on the `Response` instance.\nOriginal exception text was: 'Response' object has no attribute 'pool'.","Traceback (most recent call last):\n  File \"/opt/rockstor-dev/src/rockstor/rest_framework_custom/generic_view.py\", line 41, in _handle_exception\n    yield\n  File \"/opt/rockstor-dev/src/rockstor/storageadmin/views/pool_balance.py\", line 227, in post\n    return Response(PoolBalanceSerializer(ps).data)\n  File \"/opt/rockstor-dev/eggs/djangorestframework-3.9.3-py2.7.egg/rest_framework/serializers.py\", line 563, in data\n    ret = super(Serializer, self).data\n  File \"/opt/rockstor-dev/eggs/djangorestframework-3.9.3-py2.7.egg/rest_framework/serializers.py\", line 262, in data\n    self._data = self.to_representation(self.instance)\n  File \"/opt/rockstor-dev/eggs/djangorestframework-3.9.3-py2.7.egg/rest_framework/serializers.py\", line 517, in to_representation\n    attribute = field.get_attribute(instance)\n  File \"/opt/rockstor-dev/eggs/djangorestframework-3.9.3-py2.7.egg/rest_framework/relations.py\", line 192, in get_attribute\n    return super(RelatedField, self).get_attribute(instance)\n  File \"/opt/rockstor-dev/eggs/djangorestframework-3.9.3-py2.7.egg/rest_framework/fields.py\", line 474, in get_attribute\n    raise type(exc)(msg)\nAttributeError: Got AttributeError when attempting to get a value for field `pool` on serializer `PoolBalanceSerializer`.\nThe serializer field might be named incorrectly and not match any attribute or key on the `Response` instance.\nOriginal exception text was: 'Response' object has no attribute 'pool'.\n"]
```
i.e. a programmatic error in passing a non PoolBalance object to an inappropriate serialiser.

## Related pending issue
We have the same API behaviour when posting to our scrub command. This should be addressed in a linked, and dedicated issue.

**Comment by phillxnet** [comment]
An alternative response could be to return a 404 response to indicate that there are no prior records?

**Comment by phillxnet** [comment]
Closing as:
Fixed by #2400
